### DISCUSSION TITLE
Multi-level faceting

### DISCUSSION THREAD

**Discussion by jw5** [vega/vega-lite#8485] [Q&A]
Trying to find a way to map the results from a three factor analysis into a set of nested trellis plots:

F_a = v1, 2D trellis plot of F_b x F_c
F_a = v2, 2D trellis plot of F_b x F_c
...
F_a = vN, 2D trellis plot of F_b x F_c

Sample data, original Version:
[Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6ALQAbZskoBWOgCtY3AHaQANOChL0qZNih4ATsjETYAfXboDY5AE81GyABNkCA40J4eKnJAAqKGB+RiaMsGBOYYRi6LZg7HEAQsjwANbIBmDIeGDoYnRgKGKEyE5ZsHjo7KaIYLAYJmCorGJe0a654ehggDgEhNwA7hkAZi1NKYjoSmGoYMPcmaxKLgYVU5FK1GCMSngZ6PBeyuE7dXiskR3cw4WDuWBGsP1KsMgRyCXL4cpZAB7RU3QeAWcQoVkYVRssEAuAQFAASGTeQweHwM3CcrHgb26hmMpjA1zAiSsNjitkYxic4WGaNmLDeAEYAJwAZgADLBVGB4NxUO09mABsxahgdiJkMNhik8N8DCsdltgXMDowJJUBTy+eDFEo6A4IM4gZgcMBIKxrHpDZVSJZrHZ5DrIABfdQGjAGVKW-o7PD63DIRjUfB6U0VD56BkAJhdjmQSh5G2oIY0BuGB2yyYgWagwwpYicltgkj92cgeFsJUtCw2eRLWcgPLErFQLz0kdd2agigMvpNkG4hEtOkiUzUObzBd85MpzpTYBjpd+mc7kHQ1GoRloeyHpQhKg7pdzM6nE7rBvLld8AEdWFMvOrGFQz134HldH2AF4Zbh6NNiV5OnOC71vYJpzqmp6+GC7jZPYB71he75QNWOy1vBBrdr2UAiEuQHwQ23BiAsIaQEe+aWrYyBWGOZYVkhkBKLyqFiLOECAYBQA)

Working 2D facet over 1D facet (just trying to get the layout right, not information density). I believe this only works as the "namespace" used in the 1D facet "columns" doesn't directly conflict with the existing row/column definitions.

[Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6ALQAbZskoBWOgCtY3AHaQANOChL0qZNih4ATsjETYAfXboDY5AE81GyABNkCA40J4eKnJAAqKGB+RiaMsGBOYYRi6LZg7HEAQsjwANbIBmDIeGDoYnRgKGKEyE5ZsHjo7KaIYLAYJmCorGJe0a654ehggDgEhNwA7hkAZi1NKYjoSmGoYMPcmaxKLgYVU5FK1GCMSngZ6PBeyuE7dXiskR3cw4WDuWBGsP1KsMgRyCXL4cpZAB7RU3QeAWcQoVkYVRssEAuAQFAASGTeQweHwM3CcrHgb26hmMpjA1zAiSsNjitkYxic4WGaNmLDeAEYAJwAZgADLBVGB4NxUO09mABsxahgdiJkMNhik8N8DCsdltgXMDowJJUBTy+eDFEo6A4IM4gZgcMBIKxrHpDZVSJZrHZ5DrIABfdQG4YHbJ6U08sSsVA+MCm4YUsROS22ZBWZ1cyBogZeyDBymW2CSZ0uxxPFJejQGjAGVKW-o7PD6iBQIVOFh6Blstmu8uQFCMaj4Gt1hsG5BKHkbag58uN91Y0smxMhsO+MHubL2GN4WwlS0LDZ5NRQH1+l56FkZwcG34D-cG9DUahGWh7S06SJTMvH8fJ3zk5Od-eQBdL3wAR1YUy86qMFQ97vggeS6GOABeGTcHo7piK8Tq5uWe7vvYY5JqGlrThSC7rh+i4QVAK47GuMaKAYo5QCIh6oYOkA+gsCaYZOUARlG86EZaSi8qRYjOshSEQEhTpAA)

Multiple attempts at what I need: 1D facet over 2D facet. Nothing seems to work. Any direct use results in name conflicts. So I tried nested forms using a combination of row/column and rows/columns. Generates problematic results. Example:

[Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4KABGBEAkDODGALApgWwIaQFxUQFzwAdYsB6UgN2QHN0A6agSz0QFcAjOxge1IRQyUa6ALQAbZskoBWOgCtY3AHaQANOChL0qZNih4ATsjETYAfXboDY5AE81GyABNkCA40J4eKnJAAqKGB+RiaMsGBOYYRi6LZg7HEAQsjwANbIBmDIeGDoYnRgKGKEyE5ZsHjo7KaIYLAYJmCorGJe0a654ehggDgEhNwA7hkAZi1NKYjoSmGoYMPcmaxKLgYVU5FK1GCMSngZ6PBeyuE7dXiskR3cw4WDuWBGsP1KsMgRyCXL4cpZAB7RU3QeAWcQoVkYVRssEAuAQFAASGTeQweHwM3CcrHgb26hmMpjA1zAiSsNjitkYxic4WGaNmLDeAEYAJwAZgADLBVGB4NxUO09mABsxahgdiJkMNhik8N8DCsdltgXMDowJJUBTy+eDFEo6A4IM4gZgcMBIKxrHpDZVSJZrHZ5DrIABfdQG4YHbJ6U1ogZeyDDCliJyWsHubL2J0uxxPFJejRuj14P0BymW2zIKxqKA8sSsVAvPQMqMQA0x+BxkslyAYAypS39HZJ12VqBCpwsPTSNnNyuQFCMaj4Tvd+NV5BKHkbagVlsJrFJk3+wPB3ywSRZyA+2CF4uzqC-Gd7qDoajUIy0PaWnSRKb6o9QFNBtPLu9HyB4WwlS0AR1YUy86qMFQr57pACB5Loi4AF4ZNwejumIrxOqOla7rOkD2Iuj4rlAoYUh+G4fl+vgLBseQbooBgLlAIgHmhLaQDmCzJi+vjppmXLvp+kGaLyOzkchs6CSWwnIU6QA)

Moving on to trying to use 1D repeat on 2D facet, but there is apparently no way to provide values taken from the data, rather than hard coded as string constants. Yes, it makes sense for some of its default uses, but I'd still like to be able to dynamically select the field names to use.

Any pointers appreciated,
Jim

**Comment by domoritz** [comment]
You can specify a facet of facet but there may be some bugs (e.g. https://github.com/vega/vega-lite/issues/2761).

**Comment by jw5** [comment]
Unfortunately, that bug only considers the simple case of the product of
facets on two different dimensions, rows and columns. Thus they can be
fused.  And while the optimization of combining the display axis is nice
(ala Tableau or GGplot), it isn't strictly required.

But what I need is the next level up with say 4 factors. If you have a
given result from say facet(row=f1, column=f2), I need to use that as the
spec I need to facet over with facet(row=f3, column=f4). So you end up with
an array of arrays. GGplot can do this either by using nesting in the
single set of row column labels, or by generating multiple "standalone"
charts that are then composed into the topleve grid.

The only solution I have come up with is to hard code one level of faceting
using hconcat/vconcat to build the inner charts with appropriate hard coded
filters. But it will not be sensitive to changes in the input data factor
values :-(

Thanks,
Jim
